network/records/node_info.go

```diff
@@ -34,8 +34,14 @@ func NewNodeInfo(forkVersion forksprotocol.ForkVersion, networkID string) *NodeI
 }
 
 // Seal seals and encodes the record to be sent to other peers
-func (ni *NodeInfo) Seal(privateKey crypto.PrivKey) ([]byte, error) {
-	ev, err := record.Seal(ni, privateKey)
+func (ni *NodeInfo) Seal(privateKey crypto.PrivKey, handshakeData HandshakeData, signature []byte) ([]byte, error) {
+	signedNodeInfo := SignedNodeInfo{
+		NodeInfo:      ni,
+		HandshakeData: handshakeData,
+		Signature:     signature,
+	}
+
+	ev, err := record.Seal(signedNodeInfo, privateKey)
```

Comment:
here it should be sealed with the network Private key. only the handshakeData should be signed with the operator privatekey.